learning/modules/quantum-mechanics/getting-started-with-qiskit.ipynb

```diff
@@ -0,0 +1,1291 @@
+    "qc.cx(1, 2)\n",
+    "\n",
+    "# AND (carry) into qubit 3\n",
+    "qc.ccx(0, 1, 3)  # a AND b\n",
+    "\n",
+    "# measure\n",
+    "qc.measure_all()\n",
+    "\n",
+    "\n",
+    "qc.draw(\"mpl\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cd129de5-0b73-44db-97ea-f819184e3661",
+   "metadata": {},
+   "source": [
+    "Above is the circuit diagram for the quantum half-adder circuit. As mentioned previously, the wires represent qubits $0$ to $3$ ordered from top to bottom, and the classical bit register is the bottom double-lined wire. Then, reading from left to right, we see how the gates are applied to each qubit by seeing where the boxes show up on the corresponding wires. Finally, the measurements are shown at the end. The measurements collapse the qubit states into definite $0$ or $1$ values, and the results are sent to a classical register.\n",
+    "\n",
+    "One subtlety: although the circuit diagram is drawn left to right, when writing the corresponding matrix expression we must read it right to left. This is because in matrix multiplication, the operator closest to the state vector acts first. So, for example, the above circuit (ignoring the measurements) would be written as:\n",
+    "\n",
+    "$$\n",
+    "CCNOT(q_0,q_1,q_2)CNOT(q_1, q_2)CNOT(q_0,q_2)\\vert q_2 q_1 q_0\\rangle\n",
+    "$$\n",
+    "\n",
+    "\n",
+    "\n",
+    "#### 2. Optimizing:\n",
+    "\n",
+    "Next, we need to optimize the circuit to be run on the quantum hardware. This optimization is accomplished through the transpiler, which translates the abstract circuit shown above into instructions that the quantum computer will understand. It assigns the logical qubits above to real, physical qubits on the processor and rewrites the gates in terms of its own native set of gates that have been optimized to run on the quantum computer. Finally, the transpiler also implements something called \"error suppression and mitigation\" to try to minimize the effect of errors on the result. This isn't so important for our very simple circuit, but if you continue in your quantum computing journey to run more complicated circuits, you will soon see the value of error suppression and mitigation. If you want to learn more about this, see Olivia Lane's course, [Quantum Computing in Practice](/learning/courses/quantum-computing-in-practice/introduction).\n",
+    "\n",
+    "First, we load the packages that are required to communicate with IBM quantum computers and select a backend to run on. We can either choose the least busy backend, or select a specific backend whose properties we know.\n",
+    "\n",
+    "There is code below for saving your credentials upon first use. Be sure to delete this information from the notebook after saving it to your environment, so that your credentials are not accidentally shared when you share the notebook. See Set up your IBM Cloud account and Initialize the service in an untrusted environment for more guidance."
```

Comment:
```suggestion
    "There is code below for saving your credentials upon first use. Be sure to delete this information from the notebook after saving it to your environment, so that your credentials are not accidentally shared when you share the notebook. See [Set up your IBM Cloud account](/docs/guides/cloud-setup) and [Initialize the service in an untrusted environment](/docs/guides/cloud-setup-untrusted) for more guidance."
```